users/api_views.py

```diff
@@ -14,14 +14,39 @@
 from rest_framework_simplejwt.tokens import RefreshToken
 from rest_framework_simplejwt.views import TokenObtainPairView, TokenRefreshView
 
-from .models import User
-from .serializers import UserSerializer
+from .models import Profile, User, UserSettings
+from .serializers import ProfileSerializer, UserRegistrationSerializer, UserSerializer, UserSettingsSerializer
 from .tokens import account_activation_token
 
 
+class JWTAuthenticatedView:
+    permission_classes: ClassVar[list] = [IsAuthenticated]
+    authentication_classes: ClassVar[list] = [JWTStatelessUserAuthentication]
+
+
 class UserAPIRegistration(CreateAPIView):
+    queryset = User.objects.all()
+    serializer_class = UserRegistrationSerializer
+
+
+class UserRetrieveAPIView(JWTAuthenticatedView, RetrieveAPIView):
     queryset = User.objects.all()
     serializer_class = UserSerializer
+    lookup_field = "nickname"
+
+
+class ProfileAPIView(JWTAuthenticatedView, RetrieveAPIView, UpdateAPIView):
```

Comment:
Please change name so it is clear it is retrieve logged in user. Like CurrentUserProfileAPIView or sth.